orcid-core/src/main/java/org/orcid/core/manager/v3/impl/WorkManagerImpl.java

```diff
@@ -494,5 +518,15 @@ private WorkEntity createCopyOfUserPreferredWork(MinimizedWorkEntity preferred)
         workEntity.setWorkUrl(preferredFullData.getWorkUrl());
         return workEntity;
     }
+
```

Comment:
Please add a unit test that ensures that we store the top 50 contributors when an API request comes with more than 50; please make it a mock test so you dont need database here becase hsqldb doesnt work well with json fields